snapshot/bundlestore/groupcache.go

```diff
@@ -56,62 +61,90 @@ func MakeGroupcacheStore(underlying Store, cfg *GroupcacheConfig) (Store, http.H
 		},
 	))
 
-	return &groupcacheStore{underlying: underlying, cache: cache}, pool, nil
+	return &groupcacheStore{underlying: underlying, cache: cache, stat: stat}, pool, nil
 }
 
 // Convert 'host:port' node ids to the format expected by groupcache peering, http URLs.
-func toPeers(nodes []cluster.Node) []string {
+func toPeers(nodes []cluster.Node, stat stats.StatsReceiver) []string {
 	peers := []string{}
 	for _, node := range nodes {
 		peers = append(peers, "http://"+string(node.Id()))
 	}
 	log.Print("New groupcacheStore peers: ", peers)
+	stat.Counter("peerDiscoveryCounter").Inc(1)
+	stat.Gauge("peerCountGauge").Update(int64(len(peers)))
 	return peers
 }
 
 // Loop will listen for cluster updates and create a list of peer addresses to update groupcache.
 // Cluster is expected to include the current node.
-func loop(c *cluster.Cluster, pool *groupcache.HTTPPool) {
+func loop(c *cluster.Cluster, pool *groupcache.HTTPPool, stat stats.StatsReceiver) {
 	sub := c.Subscribe()
-	pool.Set(toPeers(c.Members())...)
+	pool.Set(toPeers(c.Members(), stat)...)
 	for {
 		select {
 		case <-sub.Updates:
-			pool.Set(toPeers(c.Members())...)
+			pool.Set(toPeers(c.Members(), stat)...)
 		}
 	}
 }
 
+func updateCacheStats(cache *groupcache.Group, stat stats.StatsReceiver) {
```

Comment:
I'm sorry, I don't understand this and it looks werd... Shouldn't updating it on the stat be enough? Why this middleman function?